### PR TITLE
acrn-config: avoid conflict slot for launch config

### PR DESCRIPTION
**Pull request by liuwei987**
- avoid conflict slot for launch config
- Kata VM is not supported on dual-core systems

    Tracked-On: #4312
    Signed-off-by: Wei Liu <weix.w.liu@intel.com>
    Acked-by: Victor Sun <victor.sun@intel.com>

### CODE REVIEW

File: misc/acrn-config/board_config/new_board_kconfig.py [hunk 1]
```diff
@@ -145,4 +145,9 @@ def generate_file(config):
     print("CONFIG_HV_RAM_START={}".format(hex(hv_start_addr)), file=config)
     print("CONFIG_HV_RAM_SIZE={}".format(hex(hv_ram_size)), file=config)
 
+    cpu_core_num = len(board_cfg_lib.get_processor_info())
+    if cpu_core_num == 2:
```
Comment (binbinwu1) [inline misc/acrn-config/board_config/new_board_kconfig.py]: just 2? How about > 2?

File: misc/acrn-config/library/launch_cfg_lib.py [hunk 2]
```diff
@@ -683,3 +684,25 @@ def cpu_sharing_check(cpu_sharing, item):
             key = "uos:id={},{}".format(vm_i, item)
             ERR_LIST[key] = "The same pcpu was configurated in scenario config, and not allow to set the cpu_sharing to 'Disabled'!"
             return
+
+
+def bdf_duplicate_check(bdf_dic):
+    """
+    Check if exist duplicate slot
+    :param bdf_dic: cotains all selected pass-through devices
```
Comment (binbinwu1) [inline misc/acrn-config/library/launch_cfg_lib.py]: typo, cotains -> contains

File: misc/acrn-config/board_config/new_board_kconfig.py (hunk 1 above)
Comment (liuwei987) [reply misc/acrn-config/board_config/new_board_kconfig.py]: Currently, KATA VM is one optional VM in SDC scenario, when we the cpu number have 2 cpu cores, they will be assigned to SOS_VM and POST VM, and there is no available cpu for KATA.
But if the cpu core > 2, there is available cpu for KATA, so support KATA VM for sdc scenario if  more than 2 cpu cores.

File: misc/acrn-config/library/launch_cfg_lib.py (hunk 2 above)
Comment (liuwei987) [reply misc/acrn-config/library/launch_cfg_lib.py]: OK, Got it.

File: misc/acrn-config/board_config/new_board_kconfig.py (hunk 1 above)
Comment (binbinwu1) [reply misc/acrn-config/board_config/new_board_kconfig.py]: OK

**Comment by HuiHuangShi** [comment]
No new violations to the coding guideline detected.<br>No New Name Conflict<br><br>No New Function Declaration/Definition Mismatch<br>

**Pull request merged**